powertools-serialization/src/test/java/software/amazon/lambda/powertools/utilities/EventDeserializerTest.java

```diff
@@ -129,7 +129,23 @@ public void testDeserializeSQSEventMessageAsObject_shouldThrowException(SQSEvent
     public void testDeserializeAPIGatewayEventAsList_shouldThrowException(APIGatewayProxyRequestEvent event) {
         assertThatThrownBy(() -> extractDataFrom(event).asListOf(Product.class))
                 .isInstanceOf(EventDeserializationException.class)
-                .hasMessageContaining("consider using 'as' instead");
+                .hasMessageContaining("consider using 'as' instead")
+                .hasMessageContaining("Cannot load the event as a list of");
+    }
+
+    @ParameterizedTest
+    @Event(value = "apigw_event.json", type = HashMap.class)
```

Comment:
apigw_event.json contains a `APIGatewayProxyRequestEvent`, not a `HashMap`... How can it even work ?!

Reply:
That was because I had as value type the `APIGatewayProxyRequestEvent`, so it was successfully converting to a maps the properties it knew, and ignoring the rest. 
That was indeed a bit confusing. I created a new json and object that is mapped to a map.